test/Microsoft.AspNetCore.Routing.Tests/Matchers/DfaMatcher.cs

```diff
@@ -65,7 +63,15 @@ public unsafe override Task MatchAsync(HttpContext httpContext, IEndpointFeature
                     }
                 }
 
-                matches.Add((candidates[i].Endpoint, values));
+                // This is some super super temporary code so we can pass the benchmarks
+                // that do HTTP method matching.
+                var httpMethodConstraint = candidate.Endpoint.Metadata.GetMetadata<HttpMethodEndpointConstraint>();
+                if (httpMethodConstraint != null && !MatchHttpMethod(httpContext.Request.Method, httpMethodConstraint))
+                {
+                    goto notmatch;
```

Comment:
Is this goto really required? Isn't it just skipping one line immediately below it?

Reply:
Dang. I try to hide one goto a week in the product and I failed 😢